Implement GetAcl/SetAcl, along with other changes

msteffen opened the pull request:
* Implement GetAcl and SetAcl
* Force the auth service to read whether it's activated from etcd on
startup
* Propagate authentication tokens from incoming context to outgoing
context in CreateRepo

jdoliner commented on src/client/auth/auth.go:
```diff
@@ -36,3 +39,19 @@ func (e NotActivatedError) Error() string {
+		mdOut[ContextTokenKey] = value
+	}
+	return metadata.NewOutgoingContext(ctx, mdOut)
+}
```
Is this function doing anything besides stripping out extraneous information like metrics? If so what? It seems like the metadata library distinguishes between incoming and outgoing messages but I can't quite understand to what end it does this.

jdoliner commented:
One clarification requested. Other than that lgtm

msteffen commented:
I believe the problem with TestUseMultipleWorkers was something I introduced when I added a backoff loop to `pachyderm_test.go:2991`. It's expecting to find the pipeline workers in a certain state in etcd, and by running that loop less frequently I reduced the likelihood that the test sees the state it's expecting
